grommet/grommet
# Typescript typing for few utils

Quick PR: I wanted to have fully customised & strongly typed grommet theme, which is easily done by copy/pasting code from `themes/base` as a starting point to my `theme.ts` file.

Few utils used there were imported from `grommet/utils` and missing typescript definitions. Feel free to reject if those are not considered as public API.

#### What does this PR do?
Add typescript typings for utils needed in `themes/base.js`

#### Where should the reviewer start?
Look at the commit :)

#### What testing has been done on this PR?
Few quick typings tests. You don't keep those in repo so they are not added.

#### How should this be manually tested?
Typescript 3.0 required. 
Build, import and then check i.e.
```ts
const object = deepMerge ({a: 1}, {b: 2}, {c: 3}) // object has correct intersection type

// or smth like that
type Expect = {
    a: number
    b: number
    c: number
}

const expect: Expect = object // no errors

// and then
// $ tsc --noEmit
``` 

#### Any background context you want to provide?
:arrow_up: 
#### What are the relevant issues?
:no_entry_sign: 
#### Screenshots (if appropriate)
:no_entry_sign: 
#### Do the grommet docs need to be updated?
:question:  - I guess not.
#### Should this PR be mentioned in the release notes?
No
#### Is this change backwards compatible or is it a breaking change?
Yes
